packages/embark-code-runner/src/index.ts

```diff
@@ -52,101 +50,57 @@ export default class CodeRunner {
+
+    // register listeners for when EmbarkJS runs registerProvider through the console.
+    // For example, when `EmbarkJS.Storage.registerProvider(...)` is run through the console,
+    // emit the `runcode:storage:providerRegistered` event, and fire any requests attached to it
+    Object.keys(EmbarkJS)
+      .filter((propName) => EmbarkJS[propName].hasOwnProperty("registerProvider"))
+      .forEach((providerName) => {
+        this.generateListener(providerName.toLowerCase(), ProviderEventType.ProviderRegistered);
+      });
 
-private registerCommands() {
-  this.events.setCommandHandler("runcode:getContext", (cb) => {
-    cb(this.vm.options.sandbox);
-  });
-  this.events.setCommandHandler("runcode:eval", this.evalCode.bind(this));
-  this.events.setCommandHandler("runcode:ready", (cb) => {
-    if (this.ready) {
-      return cb();
-    }
-    this.events.once("runcode:ready", cb);
-  });
-  this.events.setCommandHandler("runcode:embarkjs:reset", this.resetEmbarkJS.bind(this));
-
-  // register listeners for when EmbarkJS runs registerProvider through the console.
-  // For example, when `EmbarkJS.Storage.registerProvider(...)` is run through the console,
-  // emit the `runcode:storage:providerRegistered` event, and fire any requests attached to it
-  Object.keys(EmbarkJS)
-    .filter((propName) => EmbarkJS[propName].hasOwnProperty("registerProvider"))
-    .forEach((providerName) => {
-      this.generateListener(providerName.toLowerCase(), ProviderEventType.ProviderRegistered);
-    });
+    // register listeners for when EmbarkJS runs setProvider through the console.
+    // For example, when `EmbarkJS.Storage.setProvider(...)` is run through the console,
+    // emit the `runcode:storage:providerSet` event, and fire any requests attached to it
+    Object.keys(EmbarkJS)
+      .filter((propName) => EmbarkJS[propName].hasOwnProperty("setProvider"))
+      .forEach((providerName) => {
+        this.generateListener(providerName.toLowerCase(), ProviderEventType.ProviderSet);
+      });
+  }
 
-  // register listeners for when EmbarkJS runs setProvider through the console.
-  // For example, when `EmbarkJS.Storage.setProvider(...)` is run through the console,
-  // emit the `runcode:storage:providerSet` event, and fire any requests attached to it
-  Object.keys(EmbarkJS)
-    .filter((propName) => EmbarkJS[propName].hasOwnProperty("setProvider"))
-    .forEach((providerName) => {
-      this.generateListener(providerName.toLowerCase(), ProviderEventType.ProviderSet);
+  private resetEmbarkJS(cb: Callback<null>) {
+    this.events.request("code-generator:embarkjs:init-provider-code", (code: string) => {
+      this.evalCode(code, (err) => {
+        if (err) {
+          console.dir(err);
+          return;
+        }
+        this.events.request("code-generator:embarkjs:set-provider-code", (providerCode: string) => {
+          this.evalCode(providerCode, (errInitProvider, _result) => {
+          }, false);
```

Comment:
Not sure if this is by mistake or by design, but the callback for `evalCode()` doesn't call `cb()` anymore (used to be the case in line 138 prior to this commit)

Reply:
Okay I can confirm that this is definitely missing.

Causes tests to never run. Fix is in the making.

Reply:
Fixed with: https://github.com/embark-framework/embark/pull/1680/commits/ee5ad0d536f6a54f139e4e1742049866f3905dff